tools/osbuilder/image-builder/image_builder.sh

```diff
@@ -402,6 +420,27 @@ create_rootfs_image() {
 
 	info "Copying content from rootfs to root partition"
 	cp -a "${rootfs}"/* "${mount_dir}"
+
+	if [ "${SELINUX}" == "yes" ]; then
+		if [ "${AGENT_INIT}" == "yes" ]; then
+			die "Guest SELinux with the agent init is not supported yet"
+		fi
+
+		info "Labeling rootfs for SELinux"
+		selinuxfs_path="${mount_dir}${SELINUXFS}"
+		mkdir -p $selinuxfs_path
+		if mountpoint $SELINUXFS > /dev/null && \
+			chroot "${mount_dir}" command -v restorecon > /dev/null; then
+			mount -t selinuxfs selinuxfs $selinuxfs_path
+			chroot "${mount_dir}" restorecon -RF -e ${SELINUXFS} /
+			chroot "${mount_dir}" chcon -t container_runtime_exec_t "/usr/bin/${agent_bin}"
```

Comment:
Hi @ManaSugi ,

I wonder whether we should follow-up with changing the (container) SELinux policy to have the agent binary reflected in the file contexts. The `chcon` is a temporary solution -- a local permanent solution would be with `semanage fcontext`. 

However, this is actually only required when file context information get lost or if a `restorecon` is run again to correct according to the file contexts present in the policy. wdyt?

Reply:
@hbrueckner Thank you for the comment. That's good point.
Yes, `semanage fcontext` is better than `chcon`, but the best solution is that we'll add the `kata-agent` binary to https://github.com/containers/container-selinux/blob/main/container.fc so that we'll be no longer need to set `container_runtime_exec_t` to the `kata-agent` binary by hand in our builder script. ( All we have to do is just executing `restorecon`.)
I'll send a PR to https://github.com/containers/container-selinux in order to add it.
Untill the PR is merged and a new version of `container-selinux` package that includes the changes is distributed by CentOS, it would be better to use the `chcon` because `semanage fcontext -a -f container_runtime_exec_t ***` will fail with the follwing message when the new `container-selinux` already has the changes.
```
ValueError: File context for /var/yp(/.*)? already defined
```

> this is actually only required when file context information get lost or if a restorecon is run again to correct according to the file contexts present in the policy. wdyt?

Yes, I haven't assumed that users run `restorecon` again by hand and we can keep the labels of the guest rootfs by using `tar --selinux` and `cp --preserve=context`etc., when we release it.

@hbrueckner @fidencio 
Could you give me your thoughts?

Reply:
Hi @ManaSugi ,

> @hbrueckner Thank you for the comment. That's good point. Yes, `semanage fcontext` is better than `chcon`, but the best solution is that we'll add the `kata-agent` binary to https://github.com/containers/container-selinux/blob/main/container.fc so that we'll be no longer need to set `container_runtime_exec_t` to the `kata-agent` binary by hand in our builder script. ( All we have to do is just executing `restorecon`.) I'll send a PR to https://github.com/containers/container-selinux in order to add it as follows:
> 
> ```
> /usr/local/s?bin/kata-agent	--	gen_context(system_u:object_r:container_runtime_exec_t,s0)
> /usr/s?bin/kata-agent	        --	gen_context(system_u:object_r:container_runtime_exec_t,s0)
> ```

This seems to be me the best solution moving forward. Please cc me on your PR for review and LGTM.

> Yes, I haven't assumed that users run `restorecon` again by hand and we can keep the labels of the guest rootfs by using `tar --selinux` and `cp --preserve=context`etc., when we release it.

I think keeping `chcon` is an interim solution until the updated `container-sepolicy` is available. I would assume that the guest rootfs is not changed directly by users. If there is some risk, let's add a remark to the limitation in the release notes/documentation.

Reply:
@hbrueckner 

> This seems to be me the best solution moving forward. Please cc me on your PR for review and LGTM.

Awesome, I'll cc you on my PR, thank you for reviewing.

> I think keeping chcon is an interim solution until the updated container-sepolicy is available. I would assume that the guest rootfs is not changed directly by users. If there is some risk, let's add a remark to the limitation in the release notes/documentation.

That's right. I'll add a comment above the `chcon` operation in the builder script and descriptions about it to the documentation. I'll update later.

Reply:
@hbrueckner @fidencio 
I created a PR.
https://github.com/containers/container-selinux/pull/194

Reply:
@hbrueckner 
I'll add a comment above the chcon operation in the builder script and descriptions to the documentation.